source/common/ssl/connection_impl.cc

```diff
@@ -251,8 +251,22 @@ std::string ConnectionImpl::subjectPeerCertificate() {
   if (!cert) {
     return "";
   }
-  bssl::UniquePtr<char> buf(X509_NAME_oneline(X509_get_subject_name(cert.get()), nullptr, 0));
-  return std::string(buf.get());
+
+  bssl::UniquePtr<BIO> buf(BIO_new(BIO_s_mem()));
+  RELEASE_ASSERT(buf != nullptr);
+
+  // [indent=0 flags=XN_FLAG_RFC2253] is the documented set of parameters for
```

Comment:
Nit: comment seems to be wrapped at 80 chars.

Also, `indent=0` has nothing to do with the single-line, so I'm not sure if it should be included here.